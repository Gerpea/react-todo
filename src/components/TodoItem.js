import React from 'react'
import PropTypes from 'prop-types'
import { Card, Form } from 'react-bootstrap'

const TodoItem = ({ isDone, title }) => {
  return (
    <Card body>
      <Form.Check data-testid='item' type='checkbox' label={title} value={isDone} />
      {isDone && <Card.Link data-testid='delete-link'>Удалить</Card.Link>}
    </Card>
  )
}

TodoItem.propTypes = {
  isDone: PropTypes.bool,
  title: PropTypes.string,
}

TodoItem.defaultProps = {
  isDone: false,
  title: '',
}

export default TodoItem
