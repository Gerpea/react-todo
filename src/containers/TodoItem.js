import { connect } from 'react-redux'

import { markTodo } from '../state/todos/actions'

import TodoItem from '../components/TodoItem'

const mapDispatchToProps = {
  markTodo: (uid, isDone) => markTodo(uid, isDone),
}

export default connect(null, mapDispatchToProps)(TodoItem)
