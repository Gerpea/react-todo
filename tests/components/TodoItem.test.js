import React from 'react'
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import TodoItem from '../../src/components/TodoItem'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(<TodoItem />, div)
})

describe('title', () => {
  it('renders correctly', () => {
    const expected = 'Item'

    const { queryByText } = render(<TodoItem title={expected} />)

    expect(queryByText(expected)).not.toBeNull()
  })
})

describe('isDone', () => {
  it('should render unchecked by default', () => {
    const { getByTestId } = render(<TodoItem />)

    expect(getByTestId('item')).toHaveAttribute('value', 'false')
  })

  it('should render checked when true', () => {
    const expected = true

    const { getByTestId } = render(<TodoItem isDone={expected} />)

    expect(getByTestId('item')).toHaveAttribute('value', expected.toString())
  })

  it('should render unchecked when false', () => {
    const expected = false

    const { getByTestId } = render(<TodoItem isDone={expected} />)

    expect(getByTestId('item')).toHaveAttribute('value', expected.toString())
  })
})

describe('delete link', () => {
  it('should not render delete link when isDone is false', () => {
    const { queryByTestId } = render(<TodoItem isDone={false} />)

    expect(queryByTestId('delete-link')).toBeNull()
  })

  it('should render delete link when isDone is true', () => {
    const { queryByTestId } = render(<TodoItem isDone={true} />)

    expect(queryByTestId('delete-link')).not.toBeNull()
  })
})

it('matches snapshot', () => {
  const tree = renderer.create(<TodoItem isDone={true} title='Item' />).toJSON()

  expect(tree).toMatchSnapshot()
})
