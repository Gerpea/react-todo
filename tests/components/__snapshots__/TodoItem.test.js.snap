// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="form-check"
    >
      <input
        className="form-check-input"
        data-testid="item"
        disabled={false}
        type="checkbox"
        value={true}
      />
      <label
        className="form-check-label"
        title=""
      >
        Item
      </label>
    </div>
    <a
      className="card-link"
      data-testid="delete-link"
    >
      Удалить
    </a>
  </div>
</div>
`;
